Write up Content section (#37)
directus/docs: content/3.content/4.import-export.md

---
description: Learn to import and export multiple items stored as files using Directus.
---

# Import & Export

> The content module allows importing and exporting of multiple items stored as files.

Reviewer: This reads weird because of 'multiple items stored as files'. What exactly do you mean by this?

> ## Import Items

Reviewer: Please can you add more detail here. What are the requirements to import? I assume table headers need to match field names.

Reviewer: This bit is outstanding

Author: Oh you know what, I put it above the section. That''s on me. Will fix shortly

## Export Items Menu

<!-- <video autoplay playsinline muted loop controls title="Export Data to a File">
	<source src="https://cdn.directus.io/docs/v9/app-guide/imports-exports/imports-exports-20220415A/export-items-menu-20220416A.mp4" type="video/mp4" />
</video> -->

This menu provides granular control over exactly which items and fields are exported, how they are exported, and where
they are exported.

- **Format** — Choose to export items as CSV, JSON, XML, or YAML.
- **Limit** — Set the maximum number of items to be exported.
- **Export Location** — Download the export file directly to your machine or to the file library.
- **Folder** — Choose the Folder to download to (if export location is the folder library).
- **Sort Field** — Choose field to sort items by.
- **Sort Direction** — Choose to sort items in ascending or descending order.
- **Full-Text Search** — Limit exported Items to ones which matched as search results.
- **Filter** — Limit exported items with a filter.

> - **Fields** — Add, remove, and re-order the item fields that will be exported.

Reviewer: table

---
description: Learn to import and export multiple items stored as files using Directus.
---

# Import & Export

The content module allows importing and exporting of multiple items from and to files respectively.

## Import Items

<!-- TODO: Image -->

You can import JSON or CSV files into your collection as items.

To import Items from a file, navigate to the collection and click "Import / Export" in the sidebar. Click into the import search box. A file browser will open. Once a file has been selected, click on "Start Import" to import the items.

The items will now be in the collection, and the file itself will not be stored in the Directus project.

::callout{type="warning"}

> Note that the fields in the file you import must match the field IDs of the item fields of the collection you're importing into. These will otherwise be skipped.

Reviewer: ```suggestion
The field headers in the imported file must match the field keys of the collection you're importing into. Otherwise, the column will be skipped.
```
